presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py

```diff
@@ -42,25 +42,31 @@ def anonymize(self, engine_request: AnonymizerRequest) -> str:
         analyzer_results = (
             engine_request.get_analysis_results().to_sorted_unique_results(True)
         )
+
+        # loop over each analyzer result
+        # get anonymizer type class for the analyzer result (replace, redact etc.)
+        # trigger the anonymizer method on the section of the text
+        # perform the anonymization
+        # concat the anonymized string into the output string
         for analyzer_result in analyzer_results:
+            self.__validate_position_over_text(analyzer_result, text_len)
+
             transformation = engine_request.get_transformation(analyzer_result)
```

Comment:
What do you think about calling the `get_transformation` function with the entity type and not the entire analyzer_result? Something like
```python
transformation = engine_request.get_transformation(analyzer_result.entity_type)
```
or
```python
entity_type = analyzer_result.entity_type
transformation = engine_request.get_transformation(entity_type)
```

Might make the code a bit more readable. Just a suggestion though.

Reply:
done